src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -2,40 +2,64 @@
 import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useDatabaseQueueHook from '../queue/useDatabaseQueueHook';
 import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import StorageUtils from '../../../utils/storage';
 
 const useFetchPostNotificationHook = () => {
   const {localDb, refresh} = useLocalDatabaseHook();
+  const {queue} = useDatabaseQueueHook();
 
   const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
     if (!localDb) return;
-    const allPromises: Promise<void>[] = notifications.map((notification) => {
-      const channelList = fromNotificationAPI(notification);
-      return channelList.save(localDb);
+
+    notifications.map((notification) => {
+      queue.addJob({
+        label: 'save post notifications',
+        task: async () => {
+          const channelList = fromNotificationAPI(notification);
+          await channelList.save(localDb);
+        }
+      });
+
+      return null;
     });
 
-    try {
-      await Promise.all(allPromises);
-      refresh('channelList');
-    } catch (e) {
-      console.log('error on saving notifications:', e);
-    }
+    queue.addJob({
+      label: 'refresh channelList',
+      task: async () => {
+        refresh('channelList');
+      }
+    });
   };
 
   const getAllSignedPostNotifications = async () => {
+    const previousTimestamp = StorageUtils.signedNotificationTimeStamp.get();
+
     try {
-      const signedPostNotifications = await SignedMessageRepo.getAllSignedPostNotifications();
-      saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
+      const signedPostNotifications = await SignedMessageRepo.getAllSignedPostNotifications(
+        previousTimestamp as string
+      );
+      const timeStamp = new Date().toISOString();
+      await saveNotifications(signedPostNotifications, ChannelList.fromSignedPostNotificationAPI);
+      StorageUtils.signedNotificationTimeStamp.set(timeStamp);
     } catch (e) {
       console.log('error on getting signedPostNotifications:', e);
     }
   };
 
   const getAllAnonymousPostNotifications = async () => {
     try {
+      const previousTimestamp = StorageUtils.anonymousNotificationTimeStamp.get();
       const anonymousPostNotifications =
-        await AnonymousMessageRepo.getAllAnonymousPostNotifications();
-      saveNotifications(anonymousPostNotifications, ChannelList.fromAnonymousPostNotificationAPI);
+        await AnonymousMessageRepo.getAllAnonymousPostNotifications(previousTimestamp as string);
+      await saveNotifications(
+        anonymousPostNotifications,
+        ChannelList.fromAnonymousPostNotificationAPI
+      );
+      // update timestamp
+      const timestamp = new Date().toISOString();
+      StorageUtils.anonymousNotificationTimeStamp.set(timestamp);
     } catch (e) {
       console.log('error on getting anonymousPostNotifications', e);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved here. Instead of just logging the error, consider throwing it or handling it in a way that the caller can respond to it. Also, it would be better to provide more context about the error, for example, by including the notification id or other relevant information.



<!-- This is an auto-generated comment by OSS CodeRabbit -->